### PR TITLE
Fix version name can be empty string

### PR DESCRIPTION
**Pull request by MagnusHJensen**
**Description:**
Validates the version name, to check if trimming the name results in the string being empty.
Also trims the name on insert, to remove whitespace before and after name.

**Media:**

https://user-images.githubusercontent.com/32039051/218306547-d29a6ef7-9211-401c-bc9c-8e4c83617a1f.mov

Closes: #169

### CODE REVIEW

**Comment by MagnusHJensen** [comment]
The same validation should now run on project/version creation, and project/version editing.

File: src/routes/version_creation.rs [hunk 1]
```diff
@@ -275,7 +278,7 @@ async fn version_create_inner(
                         file_name: None,
                     })
                     .collect::<Vec<_>>();
-
+                println!("{}", version_create_data.version_title.clone());
```
Comment (triphora) [inline src/routes/version_creation.rs:281]: Debug line left in
```suggestion

```

**Review by triphora — CHANGES_REQUESTED** [review]

File: src/routes/version_creation.rs (hunk 1 above)
Comment (MagnusHJensen) [reply src/routes/version_creation.rs]: Fixed

**Review by triphora — APPROVED** [review]

**Review by Geometrically — CHANGES_REQUESTED** [review]
It looks like this only applies to version editing, not creation

**Comment by MagnusHJensen** [comment]
It also applies to version creation, and assuming project creation due to no validation/trimming.
Just tested it creating a version with blank name with spaces, and it was allowed.

**Review by Geometrically — APPROVED** [review]

File: src/util/validate.rs [hunk 2]
```diff
@@ -97,3 +97,30 @@ pub fn validate_url(value: &str) -> Result<(), validator::ValidationError> {
 
     Ok(())
 }
+
+pub fn validate_name(value: &str) -> Result<(), validator::ValidationError> {
+    if value.trim().is_empty() {
+        return Err(validator::ValidationError::new(
+            "Name can not contain only whitespace.",
```
Comment (triphora) [inline src/util/validate.rs:104]:
```suggestion
            "Name cannot contain only whitespace.",
```

**Pull request merged**